src/poetry/core/packages/constraints/union_constraint.py

```diff
@@ -1,45 +1,43 @@
-from typing import TYPE_CHECKING
 from typing import Tuple
 from typing import Union
+from typing import cast
 
 from poetry.core.packages.constraints.base_constraint import BaseConstraint
 from poetry.core.packages.constraints.constraint import Constraint
 from poetry.core.packages.constraints.empty_constraint import EmptyConstraint
 from poetry.core.packages.constraints.multi_constraint import MultiConstraint
 
 
-if TYPE_CHECKING:
-    from poetry.core.packages.constraints import ConstraintTypes  # noqa
-
-
 class UnionConstraint(BaseConstraint):
-    def __init__(self, *constraints: Constraint) -> None:
+    def __init__(self, *constraints: BaseConstraint) -> None:
         self._constraints = constraints
 
     @property
-    def constraints(self) -> Tuple[Constraint]:
+    def constraints(self) -> Tuple[BaseConstraint, ...]:
         return self._constraints
 
     def allows(
-        self, other: Union[Constraint, MultiConstraint, "UnionConstraint"]
+        self,
+        other: "BaseConstraint",
     ) -> bool:
         for constraint in self._constraints:
             if constraint.allows(other):
                 return True
 
         return False
 
-    def allows_any(self, other: "ConstraintTypes") -> bool:
+    def allows_any(self, other: "BaseConstraint") -> bool:
         if other.is_empty():
             return False
 
         if other.is_any():
             return True
 
-        if isinstance(other, Constraint):
-            constraints = [other]
-        else:
+        constraints: Tuple[BaseConstraint, ...]
```

Comment:
Can you explain why this is needed here?

Reply:
seems to be not needed, I expect I refactored at some point and missed that this could be removed.